src/pytest_elasticsearch/factories.py

```diff
@@ -77,15 +97,68 @@ def elasticsearch_proc(executable='/usr/share/elasticsearch/bin/elasticsearch',
     :param str elasticsearch_logsdir: path for logs.
     :param elasticsearch_logsdir: path for elasticsearch logs
     """
+    def command_from(version):
+        """
+        Get command to run elasticsearch binary based on the version.
+
+            :param tuple version elasticsearch version
+        """
+        if version < ('2', '0', '0'):
+            return '''
+                {deamon} -p {pidfile}
+                --http.port={port}
+                --path.home={home_path}
+                --transport.tcp.port={transport_tcp_port}
+                --default.path.logs={logs_path}
+                --default.path.work={work_path}
+                --default.path.data={work_path}
+                --default.path.conf={conf_path}
+                --cluster.name={cluster}
+                --network.publish_host='{network_publish_host}'
+                --index.store.type={index_store_type}
+                --discovery.zen.ping.multicast.enabled={multicast_enabled}
+            '''
+        elif version < ('3', '0', '0'):
+            return '''
+                {deamon} -p {pidfile}
+                --http.port={port}
+                --path.home={home_path}
+                --transport.tcp.port={transport_tcp_port}
+                --default.path.logs={logs_path}
+                --default.path.work={work_path}
+                --default.path.data={work_path}
+                --default.path.conf={conf_path}
+                --cluster.name={cluster}
+                --network.publish_host='{network_publish_host}'
+                --index.store.type={index_store_type}
+            '''
+        # it is known to work for 5.x.x; 6.x.x;
+        elif version <= ('6', '2', '3'):
```

Comment:
if we need to check it here, let's have it <7 otherwise we'll be buping version every minor or patch release.